### PR TITLE
Create the Snapshot module

### PR DESCRIPTION
**Pull request by poncovka**
Create a new module for the kickstart command `snapshot` and use it in UI when possible.

### CODE REVIEW

File: pyanaconda/modules/storage/snapshot/device.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        raise KickstartParseError(_("Snapshot: origin \"%(origin)s\" of snapshot "
+                                    "\"%(name)s\" is not a valid thin LV device.")
+                                  % {"origin": request.origin, "name": request.name},
+                                  lineno=request.lineno)
+
+    if devicetree.get_device_by_name("%s-%s" % (origin_dev.vg.name, snap_name)):
+        raise KickstartParseError(_("Snapshot %s already exists.") % request.name,
+                                  lineno=request.lineno)
+    try:
+        return LVMLogicalVolumeDevice(
```
Comment (poncovka) [inline pyanaconda/modules/storage/snapshot/device.py]: @vojtechtrefny, is it safe to create this object just for a verification and then create it again when we really want to create the device? It seems to work fine.

File: tests/nosetests/pyanaconda_tests/module_snapshot_test.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        self.assertEqual(self.interface.IsRequested(SNAPSHOT_WHEN_PRE_INSTALL), False)
+        self.assertEqual(self.interface.IsRequested(SNAPSHOT_WHEN_POST_INSTALL), False)
+
+        self.module._requests = [Mock(when=SNAPSHOT_WHEN_PRE_INSTALL)]
+        self.assertEqual(self.interface.IsRequested(SNAPSHOT_WHEN_PRE_INSTALL), True)
+        self.assertEqual(self.interface.IsRequested(SNAPSHOT_WHEN_POST_INSTALL), False)
+
+        self.module._requests = [Mock(when=SNAPSHOT_WHEN_POST_INSTALL)]
+        self.assertEqual(self.interface.IsRequested(SNAPSHOT_WHEN_PRE_INSTALL), False)
+        self.assertEqual(self.interface.IsRequested(SNAPSHOT_WHEN_POST_INSTALL), True)
```
Comment (jkonecny12) [inline tests/nosetests/pyanaconda_tests/module_snapshot_test.py]: You may add test when both are set. The result should be the last wins if I'm not mistaken.

File: pyanaconda/modules/storage/snapshot/device.py (hunk 1 above)
Comment (vojtechtrefny) [reply pyanaconda/modules/storage/snapshot/device.py]: Yes, it is safe. If you don't add it to the devicetree, just creating the device doesn't affect the Blivet() instance state.

File: tests/nosetests/pyanaconda_tests/module_snapshot_test.py (hunk 2 above)
Comment (jkonecny12) [reply tests/nosetests/pyanaconda_tests/module_snapshot_test.py]: By the both I mean `SNAPSHOT_WHEN_(PRE|POST)_INSTALL` :).

Comment (poncovka) [reply tests/nosetests/pyanaconda_tests/module_snapshot_test.py]: Fixed.

**Pull request merged**